test/test_client_reset.cpp

```diff
@@ -838,4 +839,384 @@ TEST(ClientReset_PinnedVersion)
+            CHECK_EQUAL(table->get_object_with_primary_key(2).get<int64_t>("value"), 12);
+            CHECK_EQUAL(table->get_object_with_primary_key(3).get<int64_t>("value"), 13);
+        }
+        else {
+            CHECK_EQUAL(table->get_object_with_primary_key(1).get<int64_t>("value"), 0);
+            CHECK_EQUAL(table->get_object_with_primary_key(2).get<int64_t>("value"), 0);
+            CHECK_EQUAL(table->get_object_with_primary_key(3).get<int64_t>("value"), 0);
+        }
+    }
+}
+
+TEST(ClientReset_Recover_RecoveryDisabled)
+{
+    SHARED_GROUP_TEST_PATH(path_1);
+    SHARED_GROUP_TEST_PATH(path_2);
+
+    auto dbs = prepare_db(path_1, path_2, [](Transaction& tr) {
+        tr.add_table_with_primary_key("class_table", type_Int, "pk");
+    });
+    CHECK_THROW((_impl::client_reset::perform_client_reset(
+                    *test_context.logger, *dbs.first, *dbs.second, ClientResyncMode::Recover, nullptr, nullptr,
+                    {100, 200}, nullptr, [](int64_t) {}, false)),
+                _impl::client_reset::ClientResetFailed);
+    CHECK_NOT(_impl::client_reset::has_pending_reset(*dbs.first->start_read()));
+}
+
+TEST(ClientReset_Recover_ModificationsOnDeletedObject)
+{
+    SHARED_GROUP_TEST_PATH(path_1);
+    SHARED_GROUP_TEST_PATH(path_2);
+
+    ColKey col;
+    auto [db, db_fresh] = prepare_db(path_1, path_2, [&](Transaction& tr) {
+        auto table = tr.add_table_with_primary_key("class_table", type_Int, "pk");
+        col = table->add_column(type_Int, "value");
+        table->create_object_with_primary_key(1).set_all(1);
+        table->create_object_with_primary_key(2).set_all(2);
+        table->create_object_with_primary_key(3).set_all(3);
+    });
+
+    {
+        auto wt = db->start_write();
+        auto table = wt->get_table("class_table");
+        table->get_object(0).set<int64_t>(col, 11);
+        table->get_object(1).add_int(col, 10);
+        table->get_object(2).set<int64_t>(col, 13);
+        wt->commit();
+    }
+    {
+        auto wt = db_fresh->start_write();
+        auto table = wt->get_table("class_table");
+        table->get_object(0).remove();
+        table->get_object(0).remove();
+        wt->commit();
+    }
+
+    expect_reset(test_context, *db, *db_fresh, ClientResyncMode::Recover);
+
+    auto tr = db->start_read();
+    auto table = tr->get_table("class_table");
+    CHECK_EQUAL(table->size(), 1);
+    CHECK_EQUAL(table->get_object_with_primary_key(3).get<int64_t>("value"), 13);
+}
+
+SubscriptionSet add_subscription(SubscriptionStore& sub_store, const std::string& name, const Query& q,
+                                 std::optional<SubscriptionSet::State> state = std::nullopt)
+{
+    auto mut = sub_store.get_latest().make_mutable_copy();
+    mut.insert_or_assign(name, q);
+    if (state) {
+        mut.update_state(*state);
+    }
+    return mut.commit();
+}
+
+TEST(ClientReset_DiscardLocal_DiscardsPendingSubscriptions)
```

Comment:
These two tests are the ones which were failing without the change to `set_active_as_latest()`.